src/events/command-handler/CoreMessageParser.ts

```diff
@@ -1,16 +1,21 @@
 import type { PieceContext } from '@sapphire/pieces';
-import type { Message } from 'discord.js';
+import { Message, NewsChannel, Permissions, TextChannel } from 'discord.js';
 import { Event } from '../../lib/structures/Event';
 import { Events } from '../../lib/types/Events';
 
 export class CoreEvent extends Event<Events.Message> {
+	private requiredPermissions = new Permissions(['VIEW_CHANNEL', 'SEND_MESSAGES']);
 	public constructor(context: PieceContext) {
 		super(context, { event: Events.Message });
 	}
 
 	public async run(message: Message) {
-		// Stop bots from running commands.
-		if (message.author.bot) return;
+		// Stop bots and webhooks from running commands.
+		if (message.author.bot || message.webhookID) return;
+
+		// If the bot cannot run the command due to lack of permissions, return.
+		const canRun = await this.canRunInChannel(message);
+		if (!canRun) return;
```

Comment:
The fact that Sapphire has this check already makes it 300% better than Klasa LMFAO

Reply:
Actually, [it does](https://github.com/dirigeants/klasa/blob/master/src/monitors/commandHandler.ts#L17).

Reply:
Oh wat? I thought you said many times it didn't.